File: CPP/No705.cc
/**
 * Created by Xiaozhong on 2020/11/21.
 * Copyright (c) 2020/11/21 Xiaozhong. All rights reserved.
 */
#include <iostream>
#include <vector>

using namespace std;

class MyHashSet {
    vector<unsigned> hash_set;
public:
    /** Initialize your data structure here. */
    MyHashSet() {
        hash_set = vector<unsigned>(1000000 / 32, 0);
    }

    void add(int key) {
        hash_set[key >> 5] |= 1 << (key & 31);
    }

    void remove(int key) {
        hash_set[key >> 5] &= ~(1 << (key & 31));
    }

    /** Returns true if this set contains the specified element */
    bool contains(int key) {
        return hash_set[key >> 5] & (1 << (key & 31));
    }
};

File: Java/src/No705.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */
class MyHashSet {

    private int[] hashSet;

    public MyHashSet() {
        hashSet = new int[1000_001];
    }

    public void add(int key) {
        hashSet[key] = 1;
    }

    public void remove(int key) {
        hashSet[key] = 0;
    }

    /**
     * Returns true if this set contains the specified element
     */
    public boolean contains(int key) {
        return hashSet[key] == 1;
    }
}

public class No705 {

}
